db/version_builder.cc

```diff
@@ -514,10 +515,26 @@ class VersionBuilder::Rep {
     return Status::OK();
   }
 
-  void MaybeAddFile(VersionStorageInfo* vstorage, int level, FileMetaData* f) {
+  void MaybeInheritFile(VersionStorageInfo* vstorage, int level,
+                        FileMetaData* f) {
     if (levels_[level].deleted_files.count(f->fd.GetNumber()) > 0) {
       // f is to-be-deleted table file
       vstorage->RemoveCurrentStats(f);
+      deleted_base_files_[f->fd.GetNumber()] = f;
+    } else {
+      vstorage->AddFile(level, f, info_log_);
+    }
+  }
+
+  void MaybeAddFile(VersionStorageInfo* vstorage, int level, FileMetaData* f,
+                    uint64_t base_max_file_number) {
+    if (f->fd.GetNumber() <= base_max_file_number) {
+      auto it = deleted_base_files_.find(f->fd.GetNumber());
```

Comment:
why not just use `deleted_base_files_` to decide if it is a trivial move.

Reply:
To make a stronger assertion below. If it's not included in `deleted_base_files_` but have a small file number, it means we are creating a new file with an old name.